### ISSUE TITLE
Convert to new approach for Region and Culture

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/center-and-variability#617]
For https://github.com/phetsims/joist/issues/958 ...

Convert this sim to the new approach for Region and Culture.  See https://github.com/phetsims/joist/issues/958 for steps.

mean-share-and-balance should be done at the same time, since the code to be converted mostly lives in soccer-common, shared by both sims.

**Comment by pixelzoom** [comment]
After adding new SVG files for africa and africaModest (see https://github.com/phetsims/soccer-common/issues/13#issuecomment-1992492081) center-and-variability-images.json is shown below.  mean-share-and-balance-images.json is the 'usa' portion of center-and-variability-images.json.    If files are changes, or additional regions are added, then these json files will need to be modified.

 Run `grunt update` to generate the {Repo}Images.ts files.  I did not commit this because there is much more conversion required.



<details>
<summary>center-and-variability-images.json</summary>

```json
{
  "usa": {
    "kicker01Standing": "../soccer-common/images/usa/usaKicker01Standing_svg.ts",
    "kicker01PoisedToKick": "../soccer-common/images/usa/usaKicker01PoisedToKick_svg.ts",
    "kicker01Kicking": "../soccer-common/images/usa/usaKicker01Kicking_svg.ts",

    "kicker02Standing": "../soccer-common/images/usa/usaKicker02Standing_svg.ts",
    "kicker02PoisedToKick": "../soccer-common/images/usa/usaKicker02PoisedToKick_svg.ts",
    "kicker02Kicking": "../soccer-common/images/usa/usaKicker02Kicking_svg.ts",

    "kicker03Standing": "../soccer-common/images/usa/usaKicker03Standing_svg.ts",
    "kicker03PoisedToKick": "../soccer-common/images/usa/usaKicker03PoisedToKick_svg.ts",
    "kicker03Kicking": "../soccer-common/images/usa/usaKicker03Kicking_svg.ts",

    "kicker04Standing": "../soccer-common/images/usa/usaKicker04Standing_svg.ts",
    "kicker04PoisedToKick": "../soccer-common/images/usa/usaKicker04PoisedToKick_svg.ts",
    "kicker04Kicking": "../soccer-common/images/usa/usaKicker04Kicking_svg.ts",

    "kicker05Standing": "../soccer-common/images/usa/usaKicker05Standing_svg.ts",
    "kicker05PoisedToKick": "../soccer-common/images/usa/usaKicker05PoisedToKick_svg.ts",
    "kicker05Kicking": "../soccer-common/images/usa/usaKicker05Kicking_svg.ts",

    "kicker06Standing": "../soccer-common/images/usa/usaKicker06Standing_svg.ts",
    "kicker06PoisedToKick": "../soccer-common/images/usa/usaKicker06PoisedToKick_svg.ts",
    "kicker06Kicking": "../soccer-common/images/usa/usaKicker06Kicking_svg.ts",

    "kicker07Standing": "../soccer-common/images/usa/usaKicker07Standing_svg.ts",
    "kicker07PoisedToKick": "../soccer-common/images/usa/usaKicker07PoisedToKick_svg.ts",
    "kicker07Kicking": "../soccer-common/images/usa/usaKicker07Kicking_svg.ts",

    "kicker08Standing": "../soccer-common/images/usa/usaKicker08Standing_svg.ts",
    "kicker08PoisedToKick": "../soccer-common/images/usa/usaKicker08PoisedToKick_svg.ts",
    "kicker08Kicking": "../soccer-common/images/usa/usaKicker08Kicking_svg.ts",

    "kicker09Standing": "../soccer-common/images/usa/usaKicker09Standing_svg.ts",
    "kicker09PoisedToKick": "../soccer-common/images/usa/usaKicker09PoisedToKick_svg.ts",
    "kicker09Kicking": "../soccer-common/images/usa/usaKicker09Kicking_svg.ts",

    "kicker10Standing": "../soccer-common/images/usa/usaKicker10Standing_svg.ts",
    "kicker10PoisedToKick": "../soccer-common/images/usa/usaKicker10PoisedToKick_svg.ts",
    "kicker10Kicking": "../soccer-common/images/usa/usaKicker10Kicking_svg.ts",

    "kicker11Standing": "../soccer-common/images/usa/usaKicker11Standing_svg.ts",
    "kicker11PoisedToKick": "../soccer-common/images/usa/usaKicker11PoisedToKick_svg.ts",
    "kicker11Kicking": "../soccer-common/images/usa/usaKicker11Kicking_svg.ts",

    "kicker12Standing": "../soccer-common/images/usa/usaKicker12Standing_svg.ts",
    "kicker12PoisedToKick": "../soccer-common/images/usa/usaKicker12PoisedToKick_svg.ts",
    "kicker12Kicking": "../soccer-common/images/usa/usaKicker12Kicking_svg.ts",

    "kicker13Standing": "../soccer-common/images/usa/usaKicker13Standing_svg.ts",
    "kicker13PoisedToKick": "../soccer-common/images/usa/usaKicker13PoisedToKick_svg.ts",
    "kicker13Kicking": "../soccer-common/images/usa/usaKicker13Kicking_svg.ts",

    "kicker14Standing": "../soccer-common/images/usa/usaKicker14Standing_svg.ts",
    "kicker14PoisedToKick": "../soccer-common/images/usa/usaKicker14PoisedToKick_svg.ts",
    "kicker14Kicking": "../soccer-common/images/usa/usaKicker14Kicking_svg.ts",

    "kicker15Standing": "../soccer-common/images/usa/usaKicker15Standing_svg.ts",
    "kicker15PoisedToKick": "../soccer-common/images/usa/usaKicker15PoisedToKick_svg.ts",
    "kicker15Kicking": "../soccer-common/images/usa/usaKicker15Kicking_svg.ts",

    "variabilityKicker01Standing": "../soccer-common/images/usa/usaVariabilityKicker01Standing_svg.ts",
    "variabilityKicker01PoisedToKick": "../soccer-common/images/usa/usaVariabilityKicker01PoisedToKick_svg.ts",
    "variabilityKicker01Kicking": "../soccer-common/images/usa/usaVariabilityKicker01Kicking_svg.ts",

    "variabilityKicker02Standing": "../soccer-common/images/usa/usaVariabilityKicker02Standing_svg.ts",
    "variabilityKicker02PoisedToKick": "../soccer-common/images/usa/usaVariabilityKicker02PoisedToKick_svg.ts",
    "variabilityKicker02Kicking": "../soccer-common/images/usa/usaVariabilityKicker02Kicking_svg.ts",

    "variabilityKicker03Standing": "../soccer-common/images/usa/usaVariabilityKicker03Standing_svg.ts",
    "variabilityKicker03PoisedToKick": "../soccer-common/images/usa/usaVariabilityKicker03PoisedToKick_svg.ts",
    "variabilityKicker03Kicking": "../soccer-common/images/usa/usaVariabilityKicker03Kicking_svg.ts",

    "variabilityKicker04Standing": "../soccer-common/images/usa/usaVariabilityKicker04Standing_svg.ts",
    "variabilityKicker04PoisedToKick": "../soccer-common/images/usa/usaVariabilityKicker04PoisedToKick_svg.ts",
    "variabilityKicker04Kicking": "../soccer-common/images/usa/usaVariabilityKicker04Kicking_svg.ts"
  },
  "africa": {
    "kicker01Standing": "../soccer-common/images/africa/africaKicker01Standing_svg.ts",
    "kicker01PoisedToKick": "../soccer-common/images/africa/africaKicker01PoisedToKick_svg.ts",
    "kicker01Kicking": "../soccer-common/images/africa/africaKicker01Kicking_svg.ts",

    "kicker02Standing": "../soccer-common/images/africa/africaKicker02Standing_svg.ts",
    "kicker02PoisedToKick": "../soccer-common/images/africa/africaKicker02PoisedToKick_svg.ts",
    "kicker02Kicking": "../soccer-common/images/africa/africaKicker02Kicking_svg.ts",

    "kicker03Standing": "../soccer-common/images/africa/africaKicker03Standing_svg.ts",
    "kicker03PoisedToKick": "../soccer-common/images/africa/africaKicker03PoisedToKick_svg.ts",
    "kicker03Kicking": "../soccer-common/images/africa/africaKicker03Kicking_svg.ts",

    "kicker04Standing": "../soccer-common/images/africa/africaKicker04Standing_svg.ts",
    "kicker04PoisedToKick": "../soccer-common/images/africa/africaKicker04PoisedToKick_svg.ts",
    "kicker04Kicking": "../soccer-common/images/africa/africaKicker04Kicking_svg.ts",

    "kicker05Standing": "../soccer-common/images/usa/usaKicker08Standing_svg.ts",
    "kicker05PoisedToKick": "../soccer-common/images/usa/usaKicker08PoisedToKick_svg.ts",
    "kicker05Kicking": "../soccer-common/images/usa/usaKicker08Kicking_svg.ts",

    "kicker06Standing": "../soccer-common/images/usa/usaKicker14Standing_svg.ts",
    "kicker06PoisedToKick": "../soccer-common/images/usa/usaKicker14PoisedToKick_svg.ts",
    "kicker06Kicking": "../soccer-common/images/usa/usaKicker14Kicking_svg.ts",

    "kicker07Standing": "../soccer-common/images/africa/africaKicker07Standing_svg.ts",
    "kicker07PoisedToKick": "../soccer-common/images/africa/africaKicker07PoisedToKick_svg.ts",
    "kicker07Kicking": "../soccer-common/images/africa/africaKicker07Kicking_svg.ts",

    "kicker08Standing": "../soccer-common/images/usa/usaKicker12Standing_svg.ts",
    "kicker08PoisedToKick": "../soccer-common/images/usa/usaKicker12PoisedToKick_svg.ts",
    "kicker08Kicking": "../soccer-common/images/usa/usaKicker12Kicking_svg.ts",

    "kicker09Standing": "../soccer-common/images/usa/usaKicker10Standing_svg.ts",
    "kicker09PoisedToKick": "../soccer-common/images/usa/usaKicker10PoisedToKick_svg.ts",
    "kicker09Kicking": "../soccer-common/images/usa/usaKicker10Kicking_svg.ts",

    "kicker10Standing": "../soccer-common/images/usa/usaKicker02Standing_svg.ts",
    "kicker10PoisedToKick": "../soccer-common/images/usa/usaKicker02PoisedToKick_svg.ts",
    "kicker10Kicking": "../soccer-common/images/usa/usaKicker02Kicking_svg.ts",

    "kicker11Standing": "../soccer-common/images/africa/africaKicker11Standing_svg.ts",
    "kicker11PoisedToKick": "../soccer-common/images/africa/africaKicker11PoisedToKick_svg.ts",
    "kicker11Kicking": "../soccer-common/images/africa/africaKicker11Kicking_svg.ts",

    "kicker12Standing": "../soccer-common/images/usa/usaKicker04Standing_svg.ts",
    "kicker12PoisedToKick": "../soccer-common/images/usa/usaKicker04PoisedToKick_svg.ts",
    "kicker12Kicking": "../soccer-common/images/usa/usaKicker04Kicking_svg.ts",

    "kicker13Standing": "../soccer-common/images/africa/africaKicker13Standing_svg.ts",
    "kicker13PoisedToKick": "../soccer-common/images/africa/africaKicker13PoisedToKick_svg.ts",
    "kicker13Kicking": "../soccer-common/images/africa/africaKicker13Kicking_svg.ts",

    "kicker14Standing": "../soccer-common/images/africa/africaKicker14Standing_svg.ts",
    "kicker14PoisedToKick": "../soccer-common/images/africa/africaKicker14PoisedToKick_svg.ts",
    "kicker14Kicking": "../soccer-common/images/africa/africaKicker14Kicking_svg.ts",

    "kicker15Standing": "../soccer-common/images/africa/africaKicker15Standing_svg.ts",
    "kicker15PoisedToKick": "../soccer-common/images/africa/africaKicker15PoisedToKick_svg.ts",
    "kicker15Kicking": "../soccer-common/images/africa/africaKicker15Kicking_svg.ts",

    "variabilityKicker01Standing": "../soccer-common/images/africa/africaVariabilityKicker01Standing_svg.ts",
    "variabilityKicker01PoisedToKick": "../soccer-common/images/africa/africaVariabilityKicker01PoisedToKick_svg.ts",
    "variabilityKicker01Kicking": "../soccer-common/images/africa/africaVariabilityKicker01Kicking_svg.ts",

    "variabilityKicker02Standing": "../soccer-common/images/africa/africaVariabilityKicker02Standing_svg.ts",
    "variabilityKicker02PoisedToKick": "../soccer-common/images/africa/africaVariabilityKicker02PoisedToKick_svg.ts",
    "variabilityKicker02Kicking": "../soccer-common/images/africa/africaVariabilityKicker02Kicking_svg.ts",

    "variabilityKicker03Standing": "../soccer-common/images/africa/africaVariabilityKicker03Standing_svg.ts",
    "variabilityKicker03PoisedToKick": "../soccer-common/images/africa/africaVariabilityKicker03PoisedToKick_svg.ts",
    "variabilityKicker03Kicking": "../soccer-common/images/africa/africaVariabilityKicker03Kicking_svg.ts",

    "variabilityKicker04Standing": "../soccer-common/images/africa/africaVariabilityKicker04Standing_svg.ts",
    "variabilityKicker04PoisedToKick": "../soccer-common/images/africa/africaVariabilityKicker04PoisedToKick_svg.ts",
    "variabilityKicker04Kicking": "../soccer-common/images/africa/africaVariabilityKicker04Kicking_svg.ts"
  },
  "africaModest": {
    "kicker01Standing": "../soccer-common/images/africaModest/africaModestKicker01Standing_svg.ts",
    "kicker01PoisedToKick": "../soccer-common/images/africaModest/africaModestKicker01PoisedToKick_svg.ts",
    "kicker01Kicking": "../soccer-common/images/africaModest/africaModestKicker01Kicking_svg.ts",

    "kicker02Standing": "../soccer-common/images/africaModest/africaModestKicker02Standing_svg.ts",
    "kicker02PoisedToKick": "../soccer-common/images/africaModest/africaModestKicker02PoisedToKick_svg.ts",
    "kicker02Kicking": "../soccer-common/images/africaModest/africaModestKicker02Kicking_svg.ts",

    "kicker03Standing": "../soccer-common/images/africaModest/africaModestKicker03Standing_svg.ts",
    "kicker03PoisedToKick": "../soccer-common/images/africaModest/africaModestKicker03PoisedToKick_svg.ts",
    "kicker03Kicking": "../soccer-common/images/africaModest/africaModestKicker03Kicking_svg.ts",

    "kicker04Standing": "../soccer-common/images/africa/africaKicker07Standing_svg.ts",
    "kicker04PoisedToKick": "../soccer-common/images/africa/africaKicker07PoisedToKick_svg.ts",
    "kicker04Kicking": "../soccer-common/images/africa/africaKicker07Kicking_svg.ts",

    "kicker05Standing": "../soccer-common/images/africaModest/africaModestKicker05Standing_svg.ts",
    "kicker05PoisedToKick": "../soccer-common/images/africaModest/africaModestKicker05PoisedToKick_svg.ts",
    "kicker05Kicking": "../soccer-common/images/africaModest/africaModestKicker05Kicking_svg.ts",

    "kicker06Standing": "../soccer-common/images/usa/usaKicker08Standing_svg.ts",
    "kicker06PoisedToKick": "../soccer-common/images/usa/usaKicker08PoisedToKick_svg.ts",
    "kicker06Kicking": "../soccer-common/images/usa/usaKicker08Kicking_svg.ts",

    "kicker07Standing": "../soccer-common/images/africaModest/africaModestKicker07Standing_svg.ts",
    "kicker07PoisedToKick": "../soccer-common/images/africaModest/africaModestKicker07PoisedToKick_svg.ts",
    "kicker07Kicking": "../soccer-common/images/africaModest/africaModestKicker07Kicking_svg.ts",

    "kicker08Standing": "../soccer-common/images/africa/africaKicker14Standing_svg.ts",
    "kicker08PoisedToKick": "../soccer-common/images/africa/africaKicker14PoisedToKick_svg.ts",
    "kicker08Kicking": "../soccer-common/images/africa/africaKicker14Kicking_svg.ts",

    "kicker09Standing": "../soccer-common/images/africa/africaKicker03Standing_svg.ts",
    "kicker09PoisedToKick": "../soccer-common/images/africa/africaKicker03PoisedToKick_svg.ts",
    "kicker09Kicking": "../soccer-common/images/africa/africaKicker03Kicking_svg.ts",

    "kicker10Standing": "../soccer-common/images/usa/usaKicker02Standing_svg.ts",
    "kicker10PoisedToKick": "../soccer-common/images/usa/usaKicker02PoisedToKick_svg.ts",
    "kicker10Kicking": "../soccer-common/images/usa/usaKicker02Kicking_svg.ts",

    "kicker11Standing": "../soccer-common/images/africaModest/africaModestKicker11Standing_svg.ts",
    "kicker11PoisedToKick": "../soccer-common/images/africaModest/africaModestKicker11PoisedToKick_svg.ts",
    "kicker11Kicking": "../soccer-common/images/africaModest/africaModestKicker11Kicking_svg.ts",

    "kicker12Standing": "../soccer-common/images/africa/africaKicker01Standing_svg.ts",
    "kicker12PoisedToKick": "../soccer-common/images/africa/africaKicker01PoisedToKick_svg.ts",
    "kicker12Kicking": "../soccer-common/images/africa/africaKicker01Kicking_svg.ts",

    "kicker13Standing": "../soccer-common/images/africaModest/africaModestKicker13Standing_svg.ts",
    "kicker13PoisedToKick": "../soccer-common/images/africaModest/africaModestKicker13PoisedToKick_svg.ts",
    "kicker13Kicking": "../soccer-common/images/africaModest/africaModestKicker13Kicking_svg.ts",

    "kicker14Standing": "../soccer-common/images/africa/africaKicker04Standing_svg.ts",
    "kicker14PoisedToKick": "../soccer-common/images/africa/africaKicker04PoisedToKick_svg.ts",
    "kicker14Kicking": "../soccer-common/images/africa/africaKicker04Kicking_svg.ts",

    "kicker15Standing": "../soccer-common/images/africa/africaKicker15Standing_svg.ts",
    "kicker15PoisedToKick": "../soccer-common/images/africa/africaKicker15PoisedToKick_svg.ts",
    "kicker15Kicking": "../soccer-common/images/africa/africaKicker15Kicking_svg.ts",

    "variabilityKicker01Standing": "../soccer-common/images/africa/africaVariabilityKicker01Standing_svg.ts",
    "variabilityKicker01PoisedToKick": "../soccer-common/images/africa/africaVariabilityKicker01PoisedToKick_svg.ts",
    "variabilityKicker01Kicking": "../soccer-common/images/africa/africaVariabilityKicker01Kicking_svg.ts",

    "variabilityKicker02Standing": "../soccer-common/images/africaModest/africaModestVariabilityKicker02Standing_svg.ts",
    "variabilityKicker02PoisedToKick": "../soccer-common/images/africaModest/africaModestVariabilityKicker02PoisedToKick_svg.ts",
    "variabilityKicker02Kicking": "../soccer-common/images/africaModest/africaModestVariabilityKicker02Kicking_svg.ts",

    "variabilityKicker03Standing": "../soccer-common/images/africa/africaVariabilityKicker03Standing_svg.ts",
    "variabilityKicker03PoisedToKick": "../soccer-common/images/africa/africaVariabilityKicker03PoisedToKick_svg.ts",
    "variabilityKicker03Kicking": "../soccer-common/images/africa/africaVariabilityKicker03Kicking_svg.ts",

    "variabilityKicker04Standing": "../soccer-common/images/africa/africaVariabilityKicker04Standing_svg.ts",
    "variabilityKicker04PoisedToKick": "../soccer-common/images/africa/africaVariabilityKicker04PoisedToKick_svg.ts",
    "variabilityKicker04Kicking": "../soccer-common/images/africa/africaVariabilityKicker04Kicking_svg.ts"
  }
}
```

</details>

**Comment by pixelzoom** [comment]
Some thoughts on the direction for this conversion...

* Change KickerImageSet to:

```ts
export type KickerImageSet = {
  standingImageProperty: LocalizedImageProperty;
  poisedToKickImageProperty: LocalizedImageProperty;
  kickingImageProperty: LocalizedImageProperty;
};
```

* Replace KickerPortrayal and its subclasses with KickerImageSets:

```
const KickerImageSets = [
    {
      standingImageProperty: SoccerCommonImages.kicker01StandingLocalizedImageProperty,
      poisedToKickImageProperty: SoccerCommonImages.kicker01PoisedToKickLocalizedImageProperty,
      kickingImageProperty: kicker01KickingLocalizedImageProperty
    },
    ...
  ]
}
export default KickerImageSets;
```

* Change KickerNode to a single scenery Image, something like:

```ts
class KickerNode extends Image {

 public constructor( kickerImageSet: KickerImageSet, ... ) {

    const imageProperty = new DerivedProperty( [
        kicker.poseProperty
        kickerImageSet.standingImageProperty,
        kickerImageSet.poisedToKickImageProperty,
        kickerImageSet.kickingImageProperty
      ], ( pose , standingImage, poisedToKickImage, kickingImage ) => {
      if ( pose === Pose.STANDING ) {
        return standingImage;
      }
      else if ( pose === Pose.POISED_TO_KICK ) {
        return poisedToKickImage;
      }
      else {
        return kickingImage;
      }
    } );

    super( imageProperty, ... );
 }
}
```

* Since there's only 1 kicker shown at time, it's worth investigating whether that can be done with 1 instance of KickerNode, rather than the current 30 instances.  Perhaps add a KickerImageSet to each Kicker (model element) and then have KickerNode observe a `kickerProperty: TReadOnlyProperty<Kicker>'.

**Comment by marlitas** [comment]
I believe this is now ready for review. Over to @pixelzoom

**Comment by pixelzoom** [comment]
Conversion looks nice.   A couple of things in mean-share-and-balance:

Unless you're including this for future use, it's unnecessary in mean-share-and-balance-main.ts:

```ts
const preferencesModel = new PreferencesModel( {} );
```

Is this still accurate and necessary? In BalancePointSceneModel.ts:

```ts
  /**
   * Clear data is only called in the Balance Point Screen when the region and culture is changed.
   */
  public override clearData(): void {
```

**Comment by marlitas** [comment]
> Is this still accurate and necessary? In BalancePointSceneModel.ts:
Yes I actually added it because we're clearing data whenever the `RegionAndCultureProperty` changes in soccer-common, but BalancePointSceneModel adds a NumberProperty because of a number spinner that soccer-common doesn't have. It was not getting reset whenever the `RegionAndCultureProperty` was changed since the parent doesn't have it.

Removed the unnecessary `PreferencesModel`. That was vestigial.

Thanks for the review @pixelzoom. Closing.
